calculate_rankings.py

```diff
@@ -39,19 +40,40 @@ def main():
     args = parser.parse_args()
 
     date_str = args.date.replace("-", "")
+
     if args.division == "all":
         divisions = DIVISIONS  # Calculate rankings for all divisions at once
     else:
         divisions = [args.division]
 
-    setup_logger(args.output / f"calculate_rankings-{args.season}-{args.division}-{date_str}.log")
+    calculate_rankings(args.input, args.season, divisions, date_str, args.output)
+
+
+def calculate_rankings(input_path, season, divisions, date_str, output_path):
+    """
+    Perform the calculation of the rankings for the given division(s) and input.
+
+    Prerequisites:
+    * Run prepare_data.py script (use its output path as input to this script)
+
+    Outputs:
+    * CSV with Games, Summary (including ratings)
+
+    :input_path: path to the folder with all necessary files
+    :season: current year
+    :divisions: list of divisions for which the ranking should be calculated
+    :date_str: date of calculation
+    :output_path: path to the folder in which to save the output files
+    """
+    os.makedirs(output_path, exist_ok=True)
+    setup_logger(output_path / f"calculate_rankings-{season}-{'_'.join(divisions)}-{date_str}.log")
     logger = logging.getLogger("ranking.data_preparation")
 
     for division in divisions:
         dataset = GamesDataset(
-            args.input / f"{args.season}-{division}-euf-games.csv",
-            name=f"{args.season}-{division}-euf",
-            date=args.date,
+            input_path / f"{season}-{division}-euf-games.csv",
+            name=f"{season}-{division}-euf",
+            date=date_str,
```

Comment:
Here we need to have date with the separators for correct filtering of the games in the dataset, where the format with `-` is expected.

Reply:
I see, so it'd be better to take the date pre-formatted and do the formatting in this function to be safe?